components/motor/gpio/motor_encoder.go

```diff
@@ -270,40 +251,27 @@ func (m *EncodedMotor) makeAdjustments(pos, lastPos float64, now, lastTime int64
 		currentRPM = deltaPos / deltaTime
 	}
 
-	if rpmDebug {
-		m.logger.Info("making adjustments")
-		m.logger.Infof("lastPos: %v, pos: %v, goalPos: %v", lastPos, pos, m.state.goalPos)
-		m.logger.Infof("lastTime: %v, now: %v", lastTime, now)
-		m.logger.Infof("currentRPM: %v, goalRPM: %v", currentRPM, m.state.goalRPM)
-	}
+	m.logger.Debug("making adjustments")
+	m.logger.Debugf("lastPos: %v, pos: %v, goalPos: %v", lastPos, pos, goalPos)
+	m.logger.Debugf("lastTime: %v, now: %v", lastTime, now)
+	m.logger.Debugf("currentRPM: %v, goalRPM: %v", currentRPM, goalRPM)
 
 	dir := m.directionMovingInLock()
 
-	if (dir == 1 && currentRPM > m.state.goalRPM) || (dir == -1 && currentRPM < m.state.goalRPM) {
-		powerPct := m.state.lastPowerPct - (m.rampRate * dir)
-		if sign(powerPct) != dir {
-			powerPct = m.state.lastPowerPct
-		}
-		if rpmDebug {
-			m.logger.Infof("decreasing powerPct to %v", powerPct)
-		}
-		if err := m.setPower(m.cancelCtx, powerPct, true); err != nil {
-			return err
-		}
+	rpmErr := goalRPM - currentRPM
+	// adjust our power based on the error in rpm
+	newPowerPct += (m.rampRate * sign(rpmErr))
+
+	if sign(newPowerPct) != dir {
+		newPowerPct = lastPowerPct
```

Comment:
[note to self] I think what is happening is this if statement keeps the motor power from updating until after an initial direction gets set(from the kickoff `setPower` call in `goForInternal`) this also prevents the robot from driving in the wrong direction at low rpms

Reply:
agreed